gadget/mountedfilesystem_test.go

```diff
@@ -880,10 +924,32 @@ func (s *mountedfilesystemTestSuite) TestMountedUpdaterBackupSimple(c *C) {
 			c.Check(backup, testutil.FileEquals, "can't touch this")
 		}
 	}
+	// notified for both updated and new content
+	c.Check(muo.contentUpdate, DeepEquals, map[string][][]string{
+		outDir: {
+			// new file
+			{filepath.Join(s.dir, "bar"), "bar-name"},
+			// updates
+			{filepath.Join(s.dir, "foo"), "foo"},
+			{filepath.Join(s.dir, "foo"), "nested/foo"},
+		},
+	})
 
-	// running backup again does not error out
+	// running backup again (eg. after a reboot) does not error out
 	err = rw.Backup()
 	c.Assert(err, IsNil)
+	// we are notified of all files again
+	c.Check(muo.contentUpdate, DeepEquals, map[string][][]string{
+		outDir: {
+			{filepath.Join(s.dir, "bar"), "bar-name"},
+			{filepath.Join(s.dir, "foo"), "foo"},
+			{filepath.Join(s.dir, "foo"), "nested/foo"},
+			// same set of calls once more
```

Comment:
at this point I'm fine either way, we'll have to see what happens writing the boot code, but it sounded like you didn't want this behavior the last time we discussed? (or maybe I misunderstood)

Reply:
Yes. However, I think my variant could lead to a confusing state when reboot happens in between we observe the 'update' and successfully update the modeenv.